src/txmempool.cpp

```diff
@@ -1055,11 +1055,37 @@ void CTxMemPool::TrimToSize(size_t sizelimit, std::vector<COutPoint>* pvNoSpends
     }
 }
 
-bool CTxMemPool::TransactionWithinChainLimit(const uint256& txid, size_t chainLimit) const {
+uint64_t CTxMemPool::CalculateDescendantMaximum(txiter entry) const {
+    // find parent with highest descendant count
+    std::vector<txiter> candidates;
+    setEntries counted;
+    candidates.push_back(entry);
+    uint64_t maximum = 0;
+    while (candidates.size()) {
+        txiter candidate = candidates.back();
+        candidates.pop_back();
+        if (counted.find(candidate) != counted.end()) continue;
+        counted.insert(candidate);
```

Comment:
nit: should be faster if you read the return value from the insertion, rather than find & insert.

Reply:
Didn't realize `set::insert` told you if the insertion happened. Nice!